src/main/java/net/aufdemrand/denizen/events/player/PlayerConsumesScriptEvent.java

```diff
@@ -0,0 +1,115 @@
+package net.aufdemrand.denizen.events.player;
+
+import net.aufdemrand.denizen.BukkitScriptEntryData;
+import net.aufdemrand.denizen.events.BukkitScriptEvent;
+import net.aufdemrand.denizen.objects.dEntity;
+import net.aufdemrand.denizen.objects.dItem;
+import net.aufdemrand.denizen.utilities.DenizenAPI;
+import net.aufdemrand.denizen.utilities.debugging.dB;
+import net.aufdemrand.denizencore.objects.dObject;
+import net.aufdemrand.denizencore.scripts.ScriptEntryData;
+import net.aufdemrand.denizencore.scripts.containers.ScriptContainer;
+import net.aufdemrand.denizencore.utilities.CoreUtilities;
+import org.bukkit.Bukkit;
+import org.bukkit.event.EventHandler;
+import org.bukkit.event.Listener;
+import org.bukkit.event.player.PlayerItemConsumeEvent;
+
+import java.util.HashMap;
+
+public class PlayerConsumesScriptEvent extends BukkitScriptEvent implements Listener {
+
+    // <--[event]
+    // @Events
+    // player consumes item
+    // player consumes <item>

```

Comment:
Agreed, looking to do in:area for all new events as soon as you code it :)
Also, we need a meta section to describe what "area" can mean (world, cuboid, etc...)